content/TernaryOperator/TernaryOperator.q.md

```diff
@@ -0,0 +1,903 @@
+
+
+/// type=MS, answer=[2,4]
+
+Which statements correctly describe the error?
+
+ - There is no semicolon `;` at the end of the statement on line 2;
+
+ - On line 3, the second operand `Very Good` is not enclosed in double quotes `""`.
+
+ - On line 3, the colon `:` between `"Very Good"` and `"Needs Improvement"` is not allowed.
+
+ - On line 3, the ternary expression `($studentGrade > 75) ? Very Good : "Needs Improvement"` is invalid.
+
+ - On line 3, the colon `:` between `"Very Good"` and `"Needs Improvement"` should be replaced with a question mark `?`.
+
+:::
+
+
+/// type=CR, answer=[tests/TernaryOperator/MissingDoubleQuotesOnOperandTest.php]
+
+Correct the code so that it outputs the string `Needs Improvement`.
+
+```php
+<?php
+$studentGrade = 75;
+$remarks = ($studentGrade > 75) ? Very Good : "Needs Improvement";
+echo $remarks;
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+$score = 75;
+$studentGrade = $score  "Invalid value";
+echo "The student grade is: " . $studentGrade;
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is the error message?
+
+ - syntax error, unexpected `';'` on line number 3
+
+ - syntax error, unexpected `'!'` on line number 3
+
+ - syntax error, unexpected `'Good'` (T_STRING) on line number 3
+
+ - syntax error, unexpected `'$remarks'` (T_VARIABLE) on line number 3
+
+ - syntax error, unexpected `'"Invalid value"'` (T_CONSTANT_ENCAPSED_STRING) on line number 3
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the error?
+
+ - There is no semicolon `;` at the end of the statement on line 2;
+
+ - On line 3, the operand `Invalid value` is enclosed in double quotes `""`.
+
+ - On line 3, the statement `$studentGrade = $score  "Invalid value";` is invalid.
+
+ - There is an assignment operator `=` between `$studentGrade` and `$score` on line 3.
+
+ - There is no ternary operator `?:` specified between `$score` and `"Invalid value"` on line 3.
+
+:::
+
+
+/// type=CR, answer=[tests/TernaryOperator/MissingTernaryOperatorTest.php]
+
+Correct the code so that it outputs the string `The student grade is: 75`.
+
+```php
+<?php
+$score = 75;
+$studentGrade = $score  "Invalid value";
+echo "The student grade is: " . $studentGrade;
+?>
+```
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+$score = 75;
+$studentGrade = $score ?: "Invalid value";
+echo "The student grade is: " . $studentGrade;
+?>
+```
+/// type=SS, answer=[1]
+
+Remove the colon `:` from the expression `$score ?: "Invalid value"` on line 3. Execute the program. What is the error message?
+
+ - syntax error, unexpected `';'` on line number 3
+
+ - syntax error, unexpected `'!'` on line number 3
+
+ - syntax error, unexpected `'Good'` (T_STRING) on line number 3
+
+ - syntax error, unexpected `'$remarks'` (T_VARIABLE) on line number 3
+
+ - syntax error, unexpected `'"Invalid value"'` (T_CONSTANT_ENCAPSED_STRING) on line number 3
+
```

Comment:
I noticed that there is no follow up question describing the error on the previous question. Is this ok?

Reply:
Yes, this is intended since this is already explained in the previous questions and this is user discovery type of question where in they are told to perform something on the given code. And if you have noticed this is connected with the next question until the CR question. Please check. Thank you.

Reply:
Noted. Thanks.